docs/languages/csharp/async.md

```diff
@@ -2,44 +2,130 @@
 
-For those who are more theoretically-inclined, this is an implementation of the [Future/Promise concurrency model](https://en.wikipedia.org/wiki/Futures_and_promises).
+### CPU-bound Example: Performing a Calculation for a Game
+
+Say you're writing a mobile game where pressing a button can inflict damage on many enemies on the screen.  Performing the damage calcuation can be expensive, and doing it on the UI thread would make the game appear to pause as the calculation is performed!
+
+The best way to handle this is to start a background thread which does the work using `Task.Run`, and `await` its result.  This will allow the UI to feel smooth as the work is being done.
+
+```csharp
+private DamageResult CalculateDamageDone()
+{
+    // Code omitted:
+    //
+    // Does an expensive calculation and returns
+    // the result of that calculation.
+}
+
+
+calculateButton.Clicked += async (o, e) =>
+{
+    // This line will yield control to the UI CalculateDamageDone()
+    // performs its work.  The UI thread is free to perform other work.
+    var damageResult = await Task.Run(() => CalculateDamageDone());
+    DisplayDamage(damageResult);
+};
+```
+
+And that's it!  This code cleanly expresses the intent of the button's click event, it doesn't require managing a background thread manually, and it does so in a non-blocking way.
 
-A few important things to know before continuing:
+### What happens under the covers
 
-*   Async code uses `Task<T>` and `Task`, which are constructs used to model the work being done in an asynchronous context. [More on Task and Task<T>](#more-on-task-and-task-t)
-*   When the `await` keyword is applied, it suspends the calling method and yields control back to its caller until the awaited task is complete. This is what allows a UI to be responsive and a service to be elastic.
+There's a lot of moving pieces where asynchronous operations are concerned.  If you're curious about what's happening underneath the covers of `Task` and `Task<T>`, checkout the [Async in-depth](async-in-depth.md) article for more information.
+
+On the C# side of things, the compiler transforms your code into a state machine which keeps track of things like yielding execution when an `await` is reached, resuming execution when a background job has finished, and so on.
+
+For the theoretically-inclined, this is an implementation of the [Promise Model of asynchrony](https://en.wikipedia.org/wiki/Futures_and_promises).
+
+## Key Pieces to Understand
+
+*   Async code can be used for both I/O-bound and CPU-bound code, but differently for each scenario.
+*   Async code uses `Task<T>` and `Task`, which are constructs used to model work being done in the background.
+*   When the `await` keyword is applied, it suspends the calling method and yields control back to its caller until the awaited task is complete.
 *   `await` can only be used inside an async method.
```

Comment:
I don't think you've defined what the 'async' keyword does yet. That should be done before you state that 'await' can only be placed in 'async' methods.